src/ray/core_worker/core_worker.cc

```diff
@@ -2075,21 +2075,33 @@ void CoreWorker::HandleGetObjectStatus(const rpc::GetObjectStatusRequest &reques
     send_reply_callback(Status::OK(), nullptr, nullptr);
   } else {
     RAY_CHECK(owner_address.worker_id() == request.owner_worker_id());
+    bool is_freed = reference_counter_->IsPlasmaObjectFreed(object_id);
```

Comment:
Hmm apparently I added this code, but it doesn't seem necessary :) The FREED status isn't checked anymore, so maybe we can remove this? I think the important status is whether the object is out-of-scope or not.

Reply:
It is still checked in the node manager [on task reconstruction](https://github.com/ray-project/ray/blob/de5bc24c602e3ac8580e5e822b79e54fe81e2920/src/ray/raylet/node_manager.cc#L2441-L2454), so should that check be removed or is it still important to handle the `FREED` status? Are you suggesting that the `FREED` status should be absorbed into the `OUT_OF_SCOPE` status, since they're handled the same by all callers and are somewhat equivalent semantically (the object _is_ out of scope if freed)?

Also, just curious, what was the reason for waiting until the `memory_store_->GetAsync` callback fired to reply when the object has already be freed? Why not reply right away?

Reply:
Ah gotcha, okay, let's keep that then!

Hmm I don't remember the reasoning at this point, but it does reduce the number of cases that have to be handled separately in the code.

Reply:
I'm going to keep the `FREED` status for now (definitely down for removing it in a future PR), along with waiting to send the RPC reply until the callback fires. I'm guessing that it has something to do with ensuring that the memory store state is in sync with the reference counter state before returning, and I don't want to rock that boat in this PR.